src/crimson/os/seastore/cache.cc

```diff
@@ -227,7 +252,7 @@ CachedExtentRef Cache::duplicate_for_write(
   return ret;
 }
 
-std::optional<record_t> Cache::try_construct_record(Transaction &t)
+record_t Cache::try_construct_record(Transaction &t)
 {
```

Comment:
Since this is now guaranteed to succeed for a non-conflicted transaction, let's call it prepare_record instead of try_construct_record.

Reply:
done in https://github.com/ceph/ceph/pull/42080/commits/13d41f876f2dba90e7981b7a34d6c16cdc39184f